### PR TITLE
PDCT-682/only_allow_valid_url

### PR DESCRIPTION
**Pull request by asacristani**
1. Add yup validator for url
2. Add error message for url validation in Document form
3. Add the first component render test

### CODE REVIEW

**Comment by asacristani** [comment]
The test is not working yet.
But manually tested and working:
image

File: src/tests/components/forms/DocumentForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  md5_sum: '1a79a4d60de6718e8e5b326e338ae533',
+  cdn_object: 'cdn/path/to/document',
+  source_url: 'http://example.com/document',
+  content_type: 'application/pdf',
+  user_language_name: 'English',
+}
+
+// Tests
+describe('DocumentForm', () => {
+  it('validate the form incorrectly', async () => {
```
Comment (PatFawbertMills) [inline src/tests/components/forms/DocumentForm.test.tsx:56]: Suggest making this a little more specific to the test, e.g. 
"validate incorrect document URL"
and below 
"validate correct document URL"

File: src/components/forms/DocumentForm.tsx [hunk 2]
```diff
@@ -149,9 +149,12 @@ export const DocumentForm = ({
             <FormLabel>Title</FormLabel>
             <Input bg="white" {...register('title')} />
           </FormControl>
-          <FormControl isRequired>
+          <FormControl isRequired isInvalid={errors.source_url ? true : false}>
```
Comment (PatFawbertMills) [inline src/components/forms/DocumentForm.tsx:152]: Could have the shorthand `isInvalid={errors.source_url}`

**Review by PatFawbertMills — COMMENTED** [review]
Couple of minor things

File: src/components/forms/DocumentForm.tsx (hunk 2 above)
Comment (diversemix) [reply src/components/forms/DocumentForm.tsx:152]: Surely this is required even if there is no error? :shrug:

**Pull request merged**